cmd/kubeadm/app/cmd/alpha/certs.go

```diff
@@ -169,10 +170,10 @@ func newCmdGenCSR() *cobra.Command {
 
 // runGenCSR contains the logic of the generate-csr sub-command.
 func runGenCSR(config *genCSRConfig) error {
```

Comment:
Maybe we can inherit `out` from `NewCmdCertsUtility`, just like `newCmdCertsExpiration`
```go
// NewCmdCertsUtility returns main command for certs phase
func NewCmdCertsUtility(out io.Writer) *cobra.Command {
	cmd := &cobra.Command{
		Use:     "certs",
		Aliases: []string{"certificates"},
		Short:   "Commands related to handling kubernetes certificates",
	}

	cmd.AddCommand(newCmdCertsRenewal(out))
	cmd.AddCommand(newCmdCertsExpiration(out, constants.KubernetesDir))
	cmd.AddCommand(NewCmdCertificateKey())
	cmd.AddCommand(newCmdGenCSR())
	return cmd
}
```
```suggestion
func runGenCSR(out io.Writer, config *genCSRConfig) error {
```

Reply:
that is good idea. will update.